### DISCUSSION TITLE
getting TypeError: util __WEBPACK_IMPORTED_MODULE_0__.TextEncoder is not a constructor

### DISCUSSION THREAD

**Discussion by atulsilori** [vercel/next.js#72450] [Help]
### Summary

I am getting TypeError: util __WEBPACK_IMPORTED_MODULE_0__.TextEncoder is not a constructor,
I am using a custom library that exposes react component and other utilities, and it is bundled using rollup
and when i am importing a component from this custom lirbrary inside a client side component in next js (which i have to), next js version i am using is 13 and installed all node js core modules and updated next.config according to webpack 5.

**Comment by icyJoseph** [comment]
> custom library that exposes react component and other utilities

Are those utilities Node.js utilities as well?

**Reply by atulsilori** [reply]
yes, updated axios to lastest version 1.7.7 in custom library

**Reply by icyJoseph** [reply]
Alright, but it could be that, it is trying to bundle utils.TextEncoder, for client usage? or that you are importing something from this library into a client component, directly, or indirectly?

**Reply by atulsilori** [reply]
creating a bundle of custom library which contains component i require, in that bundle there is utils.TextEncoder import and yes its kind of indirect import into next js
something like

"use client"
import {Navigation} from 'custom-library'

this custom library bundle contains
import require, {TextEncoder} from 'utils'

**Reply by icyJoseph** [reply]
Right... Mmm try this in your next.config.js file:

```js
// other stuff 
webpack: (config, { nextRuntime }) => {
 if (typeof nextRuntime === "undefined") { 
      config.resolve.fallback = { 
           ...config.resolve.fallback, 
           utils: false, 
       };
  } 
  return config;
 },
 // other stuff
```

See if that gets things rolling

**Reply by atulsilori** [reply]
thanks @icyJoseph  will try this

**Reply by AyushBawane** [reply]
did it help ?
